cmake/cc_rules.cmake

```diff
@@ -235,6 +235,10 @@ endfunction()
 # Adds OBJC_FLAGS to the compile options of the given target if any of the
 # sources have filenames that indicate they are are Objective-C.
 function(add_objc_flags target)
+  if(NOT APPLE)
```

Comment:
I don't 100% understand this.  Do we still compile Objective-C for non-apple targets, but we don't want the Objective-C flags for some reason?  That's a little weird.  I'd sort of expect that instead we just don't compile it at all... but I don't have all the context I'm sure.  Maybe it's worth a comment of some kind?